### PR TITLE
Use events to notify about updates

### PR DESCRIPTION
**Pull request by tjomamokrenko**
Because `sdk` will be decoupled on different packages and they will need to receive updates from it.

Usage example (advanced):
```php
<?php

use League\Event\Emitter;
use Telegram\Bot\Api;
use Telegram\Bot\Events\UpdateReceivedEvent;

$telegram = new Api('secret');

$telegram->setEventEmitter(new Emitter());

$telegram
    ->getEventEmitter()
    ->addListener('Telegram\Bot\Events\UpdateReceivedEvent', function ($event) {
    /** @var UpdateReceivedEvent $event */

    // do something here
    assert($event->getUpdate() instanceof \Telegram\Bot\Objects\Update);

}, Emitter::P_NORMAL);

/** @var \League\Event\ListenerInterface $handleCommandsListener */
$telegram
    ->getEventEmitter()
    ->addListener('Telegram\Bot\Events\UpdateReceivedEvent', $handleCommandsListener);

/** @var \League\Event\ListenerInterface $handleConversationsListener */
$telegram
    ->getEventEmitter()
    ->addListener('Telegram\Bot\Events\UpdateReceivedEvent', $handleConversationsListener);

/** @var \League\Event\ListenerInterface $userDefinedListener */
$telegram
    ->getEventEmitter()
    ->addListener('Telegram\Bot\Events\UpdateReceivedEvent', $userDefinedListener);

// events will be emitted
$telegram->getUpdates();

// events will not be emitted
$telegram->getUpdates([], false);

// event will be emitted
$telegram->getWebhookUpdates();

// event will not be emitted
$telegram->getWebhookUpdates(false);
```

If you don't want to listen to the events:
```php
<?php

use Telegram\Bot\Api;

$telegram = new Api('secret');

// events will not be emitted, because emitter is not present
$telegram->getUpdates();

// events will not be emitted
$telegram->getUpdates([], false);

// event will not be emitted, because emitter is not present
$telegram->getWebhookUpdates();

// event will not be emitted
$telegram->getWebhookUpdates(false);
```

### CODE REVIEW

File: src/Api.php [hunk 1]
```diff
@@ -4,9 +4,10 @@
 
 use Illuminate\Contracts\Container\Container;
 use Telegram\Bot\Commands\CommandBus;
+use Telegram\Bot\Events\EmitsEvents;
+use Telegram\Bot\Events\UpdateWasReceivedEvent;
```
Comment (irazasyed) [inline src/Api.php]: It could just say `UpdateWasReceived`, No need to affix with `Event`. Since the namespace itself is different and it already gives an idea of it being an event.

File: src/Api.php [hunk 2]
```diff
@@ -4,9 +4,10 @@
...
 use Telegram\Bot\Exceptions\TelegramSDKException;
 use Telegram\Bot\FileUpload\InputFile;
-use Telegram\Bot\HttpClients\GuzzleHttpClient;
```
Comment (irazasyed) [inline src/Api.php]: Why was this removed? Not being used?

Comment (tjomamokrenko) [inline src/Api.php]: @irazasyed Yes.

**Comment by tjomamokrenko** [comment]
@irazasyed Here's an example of listener in talk package https://github.com/telegram-bot-kit/talk/blob/develop/src/Listeners/HandleConversationsListener.php

File: src/Api.php [hunk 3]
```diff
@@ -958,11 +961,17 @@ public function setWebhook(array $params)
      *
      * @return Update
      */
-    public function getWebhookUpdates()
+    public function getWebhookUpdates($emitUpdateWasReceivedEvent = true)
```
Comment (irazasyed) [inline src/Api.php]: I think we could have a global config option to enable / disable events. Because the events could be applied for things like message sending, photo sending, etc. And we can't really have params for each of them to turn on/off. It would make sense to just enable / disable from the main.

Comment (tjomamokrenko) [inline src/Api.php]: @irazasyed We need to have it, BUT the purpose of this param is the following:
Imagine if getWebhookUpdates is called twice... then events are emitted twice. What if developer does want it to happen just once? That's the purpose.

File: src/Api.php (hunk 1 above)
Comment (tjomamokrenko) [inline src/Api.php]: @irazasyed Fix coming.
